systems/registry/site/pkg/server/site.go

```diff
@@ -0,0 +1,304 @@
+	}
+
+	networkId, err := uuid.FromString(req.NetworkId)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, uuidParsingError)
+	}
+
+	instDate, err := ValidateInstallDate(req.GetInstallDate())
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	svc, err := s.networkService.GetClient()
+	if err != nil {
+		return nil, err
+	}
+
+	log.Infof("checking if network exist %s", req.NetworkId)
+	_, err = svc.Get(ctx, &npb.GetRequest{
+		NetworkId: networkId.String(),
+	})
+	if err != nil {
+		return nil, grpc.SqlErrorToGrpc(err, "network")
+	}
+
+	site := &db.Site{
+		NetworkId:     networkId,
+		Name:          req.Name,
+		BackhaulId:    backhaulId,
+		PowerId:       powerId,
+		AccessId:      accessId,
+		SwitchId:      switchId,
+		IsDeactivated: req.IsDeactivated,
+		Latitude:      req.Latitude,
+		Longitude:     req.Longitude,
+		InstallDate:   instDate.String(),
+	}
+
+	err = s.siteRepo.Add(site, func(*db.Site, *gorm.DB) error {
+		site.Id = uuid.NewV4()
+		return nil
+	})
+	if err != nil {
+		return nil, grpc.SqlErrorToGrpc(err, "site")
+	}
+	evt := &epb.AddSiteEventRequest{
+		SiteId:        site.Id.String(),
+		Name:          site.Name,
+		NetworkId:     site.NetworkId.String(),
+		IsDeactivated: site.IsDeactivated,
+		BackhaulId:    site.BackhaulId.String(),
+		PowerId:       site.PowerId.String(),
+		AccessId:      site.AccessId.String(),
+		SwitchId:      site.SwitchId.String(),
+		Latitude:      site.Latitude,
+		Longitude:     site.Longitude,
+		InstallDate:   site.InstallDate,
+	}
+
+	route := s.baseRoutingKey.SetAction("add").SetObject("site").MustBuild()
+
+	err = s.msgbus.PublishRequest(route, req)
+	if err != nil {
+		log.Errorf("Failed to publish message %+v with key %+v. Errors %s", evt, route, err.Error())
+	}
+
+	s.pushSiteCount(networkId)
+
+	return &pb.AddResponse{
+		Site: dbSiteToPbSite(site),
+	}, nil
+}
+
+func (s *SiteServer) Get(ctx context.Context, req *pb.GetRequest) (*pb.GetResponse, error) {
+	log.Infof("Getting site %s", req.SiteId)
+
+	siteId, err := uuid.FromString(req.SiteId)
+
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, uuidParsingError)
+	}
+
+	site, err := s.siteRepo.Get(siteId)
+
+	if err != nil {
+		return nil, grpc.SqlErrorToGrpc(err, "site")
+	}
+	return &pb.GetResponse{
+		Site: dbSiteToPbSite(site)}, nil
+}
+
+func (s *SiteServer) GetSites(ctx context.Context, req *pb.GetSitesRequest) (*pb.GetSitesResponse, error) {
+
+	log.Infof("Getting sites %s", req.NetworkId)
+
+	networkId, err := uuid.FromString(req.NetworkId)
+
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, uuidParsingError)
+	}
+
+	sites, err := s.siteRepo.GetSites(networkId)
+
+	if err != nil {
+		return nil, grpc.SqlErrorToGrpc(err, "site")
+	}
+	resp := &pb.GetSitesResponse{
+		Sites: dbSitesToPbSites(sites),
+	}
+
+	return resp, nil
+}
+
+func (s *SiteServer) Update(ctx context.Context, req *pb.UpdateRequest) (*pb.UpdateResponse, error) {
```

Comment:
If the goal for this update RPC is to make updating each field mandatory in order to perform an update (PUT), then this will work fine. But if the goal is to make some of these fields optional while updating (PATCH), then this will not work, because of the UUIDs validation. Empty values will fail.